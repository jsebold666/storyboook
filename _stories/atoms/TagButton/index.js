import React from 'react';
import { text, boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import readme from './README.md';
import TagButton from '../../../components/atoms/TagButton/';

const component = () => (
  <TagButton
    onClick={() => {
      action('onClick');
      alert('onClick');
    }}
    onClear={() => {
      action('onClear');
      alert('onClear');
    }}
    text={text('value', 'Clique aqui')}
    active={boolean('active', false)}
    disabled={boolean('disabled', false)}
  />
);

export default [readme, component];
