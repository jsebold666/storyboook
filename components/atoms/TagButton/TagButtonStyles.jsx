import styled, { css } from 'styled-components';
import colors from '../../../styles/colors';
// import { default as CloseIconMui } from '@material-ui/icons/Close';
// import { withStyles } from '@material-ui/core/styles';

export const TagWrapper = styled.div`
  padding: 8px 16px;
  background-color: ${colors.white};
  box-sizing: border-box;
  width: auto;
  height: 32px;
  display: inline-flex;
  align-items: center;
  border-radius: 4px;
  border: solid 1px ${colors['light-greyish']};
  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
  transition: all 350ms ease;
  flex-shrink: 0;

  ${props =>
    props.active &&
    css`
      background-color: ${colors.periwinkle};
      border-color: ${colors.periwinkle};

      & ${Text} {
        color: ${colors.white};
        margin-right: ${props => (props.text && !props.disabled ? '8px' : 0)};
      }
    `};

  ${props =>
    !props.active &&
    !props.disabled &&
    css`
      &:hover ${Text} {
        color: ${colors.periwinkle};
      }
    `};
`;

export const Text = styled.span`
  font-family: 'Rubik', 'sans-serif';
  font-size: 12px;
  font-weight: 500;
  color: ${colors['bluey-grey']};
  user-select: none;
`;

export const IconWrapper = styled.div`
  width: 16px;
  height: 16px;
  padding: 0px;
  border-radius: 50%;
  background-color: ${colors['cloudy-blue']};
`;

// export const CloseIcon = withStyles({
//   root: {
//     fontSize: '16px',
//     color: colors.white
//   }
// })(CloseIconMui);
