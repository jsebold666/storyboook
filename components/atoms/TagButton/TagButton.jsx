import React, { Component } from 'react';
import { TagWrapper, Text, IconWrapper } from './TagButtonStyles';
import CloseIcon from '../Icons/CloseIcon';

class TagButton extends Component {
  constructor() {
    super();

    this.onClickHandler = this.onClickHandler.bind(this);
    this.onClearHandler = this.onClearHandler.bind(this);
  }

  onClickHandler(event) {
    if (this.props.onClick && !this.props.disabled) this.props.onClick(event);
  }

  onClearHandler(event) {
    if (!this.props.disabled) {
      event.stopPropagation();
      if (this.props.onClear) this.props.onClear(event);
    }
  }

  render() {
    const { onClear, onClick, disabled, active, ...otherProps } = this.props;
    return (
      <TagWrapper disabled={disabled} active={active} {...otherProps} onClick={this.onClickHandler}>
        <Text>{this.props.text}</Text>

        {active &&
          !disabled && (
            <IconWrapper onClick={this.onClearHandler}>
              <CloseIcon />
            </IconWrapper>
          )}
      </TagWrapper>
    );
  }
}

export default TagButton;
