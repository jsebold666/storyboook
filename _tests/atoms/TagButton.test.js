/* globals mount */
import React from 'react';
import TagButton from '../../components/atoms/TagButton';

describe('Expect <TagButton>', () => {
  it('to render', () => {
    const props = {
      value: 'This is text',
      onChange: () => {}
    };

    const wrapper = mount(<TagButton {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('to render properly with empty', () => {
    const props = {
      value: '',
      onChange: () => {}
    };

    const wrapper = mount(<TagButton {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
