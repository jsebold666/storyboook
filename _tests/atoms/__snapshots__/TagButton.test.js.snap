// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <TagButton> to render 1`] = `
<TagButton
  onChange={[Function]}
  value="This is text"
>
  <TagButtonStyles__TagWrapper
    onChange={[Function]}
    onClick={[Function]}
    value="This is text"
  >
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "TagButtonStyles__TagWrapper-sc-1hxomxr-0",
            "isStatic": false,
            "lastClassName": "fitCAM",
            "rules": Array [
              "
  padding: 8px 16px;
  background-color: ",
              "#fff",
              ";
  box-sizing: border-box;
  width: auto;
  height: 32px;
  display: inline-flex;
  align-items: center;
  border-radius: 4px;
  border: solid 1px ",
              "#e0e7ec",
              ";
  cursor: ",
              [Function],
              ";
  transition: all 350ms ease;
  flex-shrink: 0;

  ",
              [Function],
              ";

  ",
              [Function],
              ";
",
            ],
          },
          "displayName": "TagButtonStyles__TagWrapper",
          "render": [Function],
          "styledComponentId": "TagButtonStyles__TagWrapper-sc-1hxomxr-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onChange={[Function]}
      onClick={[Function]}
      value="This is text"
    >
      <div
        className="TagButtonStyles__TagWrapper-sc-1hxomxr-0 fitCAM"
        onChange={[Function]}
        onClick={[Function]}
        value="This is text"
      >
        <TagButtonStyles__Text>
          <StyledComponent
            forwardedClass={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "TagButtonStyles__Text-sc-1hxomxr-1",
                  "isStatic": true,
                  "lastClassName": "hkcelG",
                  "rules": Array [
                    "
  font-family: 'Rubik', 'sans-serif';
  font-size: 12px;
  font-weight: 500;
  color: ",
                    "#acbec7",
                    ";
  user-select: none;
",
                  ],
                },
                "displayName": "TagButtonStyles__Text",
                "render": [Function],
                "styledComponentId": "TagButtonStyles__Text-sc-1hxomxr-1",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="TagButtonStyles__Text-sc-1hxomxr-1 hkcelG"
            />
          </StyledComponent>
        </TagButtonStyles__Text>
      </div>
    </StyledComponent>
  </TagButtonStyles__TagWrapper>
</TagButton>
`;

exports[`Expect <TagButton> to render properly with empty 1`] = `
<TagButton
  onChange={[Function]}
  value=""
>
  <TagButtonStyles__TagWrapper
    onChange={[Function]}
    onClick={[Function]}
    value=""
  >
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "TagButtonStyles__TagWrapper-sc-1hxomxr-0",
            "isStatic": false,
            "lastClassName": "fitCAM",
            "rules": Array [
              "
  padding: 8px 16px;
  background-color: ",
              "#fff",
              ";
  box-sizing: border-box;
  width: auto;
  height: 32px;
  display: inline-flex;
  align-items: center;
  border-radius: 4px;
  border: solid 1px ",
              "#e0e7ec",
              ";
  cursor: ",
              [Function],
              ";
  transition: all 350ms ease;
  flex-shrink: 0;

  ",
              [Function],
              ";

  ",
              [Function],
              ";
",
            ],
          },
          "displayName": "TagButtonStyles__TagWrapper",
          "render": [Function],
          "styledComponentId": "TagButtonStyles__TagWrapper-sc-1hxomxr-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onChange={[Function]}
      onClick={[Function]}
      value=""
    >
      <div
        className="TagButtonStyles__TagWrapper-sc-1hxomxr-0 fitCAM"
        onChange={[Function]}
        onClick={[Function]}
        value=""
      >
        <TagButtonStyles__Text>
          <StyledComponent
            forwardedClass={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "TagButtonStyles__Text-sc-1hxomxr-1",
                  "isStatic": true,
                  "lastClassName": "hkcelG",
                  "rules": Array [
                    "
  font-family: 'Rubik', 'sans-serif';
  font-size: 12px;
  font-weight: 500;
  color: ",
                    "#acbec7",
                    ";
  user-select: none;
",
                  ],
                },
                "displayName": "TagButtonStyles__Text",
                "render": [Function],
                "styledComponentId": "TagButtonStyles__Text-sc-1hxomxr-1",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="TagButtonStyles__Text-sc-1hxomxr-1 hkcelG"
            />
          </StyledComponent>
        </TagButtonStyles__Text>
      </div>
    </StyledComponent>
  </TagButtonStyles__TagWrapper>
</TagButton>
`;
